### PR TITLE
Extract hooks from side effect only components

### PR DESCRIPTION
**Pull request by timowestnosto**
Extract hooks from side effect only components to make it possible to execute the side effects both via hooks and components

### CODE REVIEW

File: src/components/NostoSession.tsx [hunk 1]
```diff
@@ -13,6 +13,16 @@ import { snakeize } from "../utils/snakeize"
  * @group Essential Functions
  */
 export default function NostoSession(props?: { cart?: Cart; customer?: Customer }) {
+  useNostoSession(props)
+  return null
```
Comment (timowestnosto) [inline src/components/NostoSession.tsx:17]: use null return to be aligned with other side effect only components

File: src/components/NostoOrder.tsx [hunk 2]
```diff
@@ -21,10 +21,17 @@ import { snakeize } from "../utils/snakeize"
  *
  * @group Components
  */
-export default function NostoOrder(props: {
-  order: Order,
-  placements?: string[]
-}) {
+export default function NostoOrder(props: { order: Order, placements?: string[] }) {
```
Comment (manikandan-ravikumar) [inline src/components/NostoOrder.tsx:24]: Maybe extract `{ order: Order, placements?: string[] }` to a type?

File: src/components/NostoProduct.tsx [hunk 3]
```diff
@@ -29,6 +29,20 @@ export default function NostoProduct(props: {
...
+/**
+ * You can personalise your product pages by using the useNostoProduct hook.
+ * 
+ * @group Hooks
+ */
+export function useNostoProduct(props: {
+  product: string
+  tagging?: Product
+  placements?: string[]
 }) {
```
Comment (manikandan-ravikumar) [inline src/components/NostoProduct.tsx:46]: extract {
  product: string
  tagging?: Product
  placements?: string[]
} to type and then we can destructure it in method param?

**Comment by manikandan-ravikumar** [comment]
Can we use type declaration and destructure pattern inside the component tsx files?

**Review by manikandan-ravikumar — APPROVED** [review]

**Pull request merged**